src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -94,6 +94,106 @@ get_iceberg_committed_offset(const iceberg::table_metadata& table) {
     }
     return std::nullopt;
 }
+
+checked<iceberg::struct_value, file_committer::errc> build_partition_key_struct(
+  const model::topic& topic,
```

Comment:
Nice. Maybe it's worth using a prefix logger that includes the topic and table, if we expect more stuff to be reused by the DLT

Reply:
I'll leave it as is for now.